agent/agentscripts/agentscripts.go

```diff
@@ -238,6 +263,13 @@ func (r *Runner) run(ctx context.Context, script codersdk.WorkspaceAgentScript)
 	cmd.WaitDelay = 10 * time.Second
 	cmd.Cancel = cmdCancel(cmd)
 
+	// Expose env vars that can be used in the script for storing data
+	// and binaries. In the future, we may want to expose more env vars
+	// for the script to use, like CODER_SCRIPT_DATA_DIR for persistent
+	// storage.
+	cmd.Env = append(cmd.Env, "CODER_SCRIPT_DATA_DIR="+scriptDataDir)
+	cmd.Env = append(cmd.Env, "CODER_SCRIPT_BIN_DIR="+r.ScriptBinDir())
```

Comment:
Will bin dir always be a subpath of data dir?

Reply:
@johnstcn yes, that's the intent. All script-data (inc. bin) will be contained in a single path for tidyness.

Reply:
Do we need both environment variables then?

Reply:
The second is not static, it's `datadir/uuid`.

Reply:
Ah, that makes much more sense then 👍